### ISSUE TITLE
Would be great if multiline strings were categorized differently

### ISSUE THREAD

**Issue by basarat** [Microsoft/TypeScript#2005]
As it stands, the default classification looks *horrible* 

![image](https://cloud.githubusercontent.com/assets/874898/6140095/c0797dc8-b1e7-11e4-9d7b-21ad2430fcc0.png)

I would be happy if there was just `InMultiLineString` (notice I don't say `Literal`) + it gets classified as  `MultilineString` 

Current: 

```ts
    const enum EndOfLineState {
        Start = 0,
        InMultiLineCommentTrivia = 1,
        InSingleQuoteStringLiteral = 2,
        InDoubleQuoteStringLiteral = 3,
    }
    enum TokenClass {
        Punctuation = 0,
        Keyword = 1,
        Operator = 2,
        Comment = 3,
        Whitespace = 4,
        Identifier = 5,
        NumberLiteral = 6,
        StringLiteral = 7,
        RegExpLiteral = 8,
    }
```

Note: feel free to call it something other than `MultiLineString` but give us *something*.

**Comment by DanielRosenwasser** [comment]
#1744 has a fix for this.

**Comment by basarat** [comment]
@DanielRosenwasser perhaps you can quickly tell me what `classifyKeywordsInGenerics` is supposed to do. Just a hint on if I should keep it `true` or `false`

**Comment by basarat** [comment]
@DanielRosenwasser also is there some public API endpoint to get the `syntacticClassifier`?

**Comment by DanielRosenwasser** [comment]
@basarat if you'll see my change, it's been renamed to `syntacticClassifierAbsent`.

Before the syntactic classifier was a thing, the lexical classifier had to do a bunch of hacky sort of tricks to try to emulate correct behavior.

Basically, you don't want to accidentally classify `number` as a keyword as a type parameter when it might just be a plain identifier after a less-than token. The syntactic classifier will overwrite this to be the correct thing (i.e. an identifier) and so you'll get flashing. However, if there *is no* syntactic classifier, the lexical classifier should use its awful tricks again.

Actually, you're using web workers for Atom right? Are you using the other classifiers?

**Comment by basarat** [comment]
> you're using web workers for Atom right?

Nope. A child process : https://github.com/TypeStrong/atom-typescript/blob/master/CONTRIBUTING.md#architecture . Node-webworker's need C/C++ toolchain on the client to install so I just went with a child process to make it easier to install. 

> Are you using the other classifiers?

nope. Just a lexical classifier : https://github.com/TypeStrong/atom-typescript/blob/master/lib/main/atom/typescriptGrammar.ts#L47

**Comment by basarat** [comment]
Reason for not using webworkers : https://github.com/atom/atom-shell/issues/797

**Comment by DanielRosenwasser** [comment]
I just realized, you're asking for a multiline string classification. What if it's just a string literal classification but it's done correctly?

**Comment by basarat** [comment]
In a single line it gets identitified as an identifier : 

![image](https://cloud.githubusercontent.com/assets/874898/6158276/f4ea9848-b29c-11e4-9043-e6a49294d638.png)

In mutliple line it gets identified as code as there is no `finalLexState` for stuff with "`" 

![image](https://cloud.githubusercontent.com/assets/874898/6158337/54e74494-b29d-11e4-9399-b9ea087723f2.png)

So for both of these I would like "just a string literal classification but it's done correctly".

**Comment by basarat** [comment]
@DanielRosenwasser fixed? Or wont-fix :heart:

**Comment by DanielRosenwasser** [comment]
If we do this I don't have to buy you chocolates for Valentine's Day. :heart:

(Will fix, but unfortunately just for 1.5: #2026)

**Comment by DanielRosenwasser** [comment]
The fix is in `master`.
